upstairs/src/downstairs.rs

```diff
@@ -1961,17 +2036,51 @@ impl Downstairs {
     /// empty).
     ///
     /// # Panics
-    /// If `self.reconcile_current_work` is not `None`
+    /// If `self.reconcile_current_work` is not `None`, or if `self.reconcile`
+    /// is `None`.
     pub(crate) async fn send_next_reconciliation_req(&mut self) -> bool {
         assert!(self.reconcile_current_work.is_none());
+
+        let Some(ref mut reconcile) = self.reconcile else {
```

Comment:
Nit: I find the `ref mut` syntax weird and would suggest
```suggestion
        let Some(reconcile) = &mut self.reconcile else {
```

Reply:
36c7c0c